Smpl+d body models

charsyme opened the pull request:
This branch contains code for downloading a large number of human models in SMPL+D body model format, as well as code for making them able to be transferred to Webots for simulations.

charsyme commented:
@ad-daniel do you know why the tests are being skipped?

ad-daniel commented:
Hi, the test aren't starting because you need to set the corresponding labels

charsyme commented:
@ad-daniel may the variable APACHE2_LICENSE_C in tests/test_license.py be incorrectly set?

charsyme commented:
@ad-daniel may the variable APACHE2_LICENSE_C in tests/test_license.py be incorrectly set? I have appended the corresponding section regarding Apache licence but I still get an error.

ad-daniel commented:
Should be fine now, I think it doesn't hurt to add the apache 2 license to the utilities as well.

passalis commented:
Hi, the PR is marked as draft. Is this indeed work in progress or we should review this PR?

charsyme commented:
> Hi, the PR is marked as draft. Is this indeed work in progress or we should review this PR?

I have finalized the code, once the data needed for the utility are up to the FTP server in Monday, I will make the PR ready for review.

charsyme commented:
The PR is ready for review.

ad-daniel commented on projects/simulation/SMPL+D_body_models/README.md:
```diff
@@ -0,0 +1,60 @@
+
+### Download the SMPL+D models and convert them as FBX (Filmbox) files. <br/>In this case the pose blend shapes are not applied.
+
+- Download the data by running:
```
Should probably mention how big this file is so users are aware.

charsyme replied:
> Should probably mention how big this file is so users are aware.

I added their size in the description

ad-daniel commented on projects/simulation/SMPL+D_body_models/README.md:
````diff
@@ -0,0 +1,60 @@
+$BLENDER_PATH/blender -P src/generate_models.py
+```
+
+### Install a demo Webots project for animating SMPL+D models. <br/> FBX models must have been previously generated.
````
```suggestion
### Install a demo Webots project for animating SMPL+D models

**Note:** FBX models must have been previously generated
```

ad-daniel commented on projects/simulation/SMPL+D_body_models/README.md:
```diff
@@ -0,0 +1,60 @@
+  <img src="./examples/model_2.png" width=150 />
+</p>
+
+### Download the raw SMPL+D models only.
```
```suggestion
### Download the raw SMPL+D models only

```

ad-daniel commented on projects/simulation/SMPL+D_body_models/README.md:
````diff
@@ -0,0 +1,60 @@
+python download_data.py raw
+```
+
+### Download the SMPL+D models and convert them as FBX (Filmbox) files. <br/>In this case the pose blend shapes are not applied.
````
```suggestion
### Download the SMPL+D models and convert them to FBX (Filmbox)

**Note:** in this case the pose blend shapes are not applied.

```

ad-daniel commented on projects/simulation/SMPL+D_body_models/README.md:
````diff
@@ -0,0 +1,60 @@
+SMPL_f_unityDoubleBlends_lbs_10_scale5_207_v1.0.0.fbx
+SMPL_m_unityDoubleBlends_lbs_10_scale5_207_v1.0.0.fbx
+```
+- Place these two files in the ```model``` directory.
````
No model directory appears to be included. So should probably mention at what level it should be placed.

charsyme replied:
The 'model' folder is created once the download_data.py has been run. 
https://github.com/tasostefas/opendr_internal/blob/ae7e4a3527d290a3b7524e2af35a24da55874097/projects/simulation/SMPL%2BD_body_models/src/download_data.py#L61

ad-daniel reviewed (commented):
I'm still in the process of checking (still downloading...), but have a few comments already

- Missing docs page and a reference to it in the `index.md`. You can probably repurpose the README for this (renaming it). Perhaps providing additional info, so far it isn't clear to me the difference between the 3 sections you made in the readme. Are they alternative approaches? Do you need to do all three one after the next?

In short, I think it could use a bit more details, the sections should be roughly:
- Intro
- What does this tool provide?
- What are the necessary steps to use it? (the info is there, but it's unclear if you need all 3, either of them, ...). 
- Are there any parameters we can change? (I don't believe so, but if it's the case they should be documented)
- Demo: all necessary steps on how to run it

ad-daniel commented on projects/simulation/SMPL+D_body_models/README.md:
```diff
@@ -0,0 +1,68 @@
+
+**Note:** FBX models must have been previously generated
+
+-  Download a database from AMASS (https://amass.is.tue.mpg.de/download.php)
```
```suggestion
-  Download ACCAD database from AMASS (https://amass.is.tue.mpg.de/download.php)

```

ad-daniel replied:
Better specify something that is known to work. I tried a different one since ACCAD wasn't working and got an error because it was missing `mocap_framerate`.
Eventually I could download ACCAD though.

charsyme replied:
Ok, I added an if-statement for mocap_framerate. In its current state the mocap_framerate isn't used by the webots controller.  However, since I haven't checked all databases, it could be safer to specify it for the ACCAD database.

ad-daniel reviewed (approved):
Thank you, only a few details from my part

passalis reviewed (commented):
Thank you! This also seems fine from my side. Can you @charsyme check the remaining issues and either accept or respond to the final minor points?

charsyme commented:
> Thank you! This also seems fine from my side. Can you @charsyme check the remaining issues and either accept or respond to the final minor points?

I believe I will address all the remaining issues today.

charsyme commented on projects/simulation/SMPL+D_body_models/src/generate_models.py:
```diff
@@ -0,0 +1,278 @@
+
+if __name__ == '__main__':
+
+    dir_in = './human_data'
```
Yes you are right, I think its ok now. I changed also L47 of download_data to avoid any conflicts.

charsyme commented on projects/simulation/SMPL+D_body_models/README.md:
````diff
@@ -0,0 +1,68 @@
+- Install the necessary packages to Blender's Python using pip (example using Blender 2.93 and Python 3.9):
+```
+$BLENDER_PATH/2.93/python/bin/python3.9 -m ensurepip
+$BLENDER_PATH/2.93/python/bin/python3.9 -m pip install numpy opencv-python opencv-contrib-python scipy
````
I added tqdm in L44 of README.md

charsyme commented:
> I'm still in the process of checking (still downloading...), but have a few comments already
> 
>     * Missing docs page and a reference to it in the `index.md`. You can probably repurpose the README for this (renaming it). Perhaps providing additional info, so far it isn't clear to me the difference between the 3 sections you made in the readme. Are they alternative approaches? Do you need to do all three one after the next?
> 
> 
> In short, I think it could use a bit more details, the sections should be roughly:
> 
>     * Intro
> 
>     * What does this tool provide?
> 
>     * What are the necessary steps to use it? (the info is there, but it's unclear if you need all 3, either of them, ...).
> 
>     * Are there any parameters we can change? (I don't believe so, but if it's the case they should be documented)
> 
>     * Demo: all necessary steps on how to run it

@ad-daniel I hope it is more clear now. I have added more clarifications to README.md. I also addressed the missing doc in references/docs and I linked it to index.md. In addition I did a minor change in the name of the utility's folder.

ad-daniel commented:
@charsyme currently the `download_data` script doesn't appear to be working anymore

```
$ python download_data.py raw

Downloading data from ftp://opendrdata.csd.auth.gr/simulation/SMPLD_body_models/human_models.tar.gz to /home/daniel/opendr_internal/projects/simulation/SMPL+D_body_models/human_models.tar.gz
Traceback (most recent call last):
  File "download_data.py", line 75, in <module>
    download_data(raw_data)
  File "download_data.py", line 51, in download_data
    urlretrieve(human_data_url, downloaded_human_data_path, reporthook=reporthook)
  File "/usr/lib/python3.8/urllib/request.py", line 257, in urlretrieve
    tfp = open(filename, 'wb')
FileNotFoundError: [Errno 2] No such file or directory: '/home/daniel/opendr_internal/projects/simulation/SMPL+D_body_models/human_models.tar.gz'
```

charsyme commented:
@ad-daniel There were a few minor issues with the recent changes I had made. I have checked all the steps again and it worked fine locally.

ad-daniel reviewed (approved):
Worked for me as well, thank you!

passalis reviewed (approved):
This also seems fine for me as well. Thank you!
